Makefile gives all-recursive error

I'm compiling the code from git, but I get [this](https://pastebin.com/p9rwnhtQ) error (basically cmake closes due to recursion error, which is caused by `extern "C"` in Dock.cpp file). I can't find any resource that talks about how to fix this, and it doesn't look like a missing dependency on my system (I'm on Arch but I need to compile it because running `yay -S` gives me this:

```
...
==> Starting pkgver()...
==> Updated version: xfce4-docklike-plugin-git 20210228-1
==> Sources are ready.
==> Making package: xfce4-docklike-plugin-git 20210228-1 (Sun Apr  4 14:19:20 2021)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting pkgver()...
==> Removing existing $pkgdir/ directory...
==> Starting build()...
Making all in src
make[1]: Entering directory '/home/goddard/.cache/yay/xfce4-docklike-plugin-git/src/docklike-plugin/src'
  CXX      libdocklike_la-Hotkeys.lo
libtool: Version mismatch error.  This is libtool 2.4.6, but the
libtool: definition of this LT_INIT comes from libtool 2.4.6.42-b88ce-dirty.
libtool: You should recreate aclocal.m4 with macros from libtool 2.4.6
libtool: and run autoconf again.
make[1]: *** [Makefile:641: libdocklike_la-Hotkeys.lo] Error 63
make[1]: Leaving directory '/home/goddard/.cache/yay/xfce4-docklike-plugin-git/src/docklike-plugin/src'
make: *** [Makefile:480: all-recursive] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
error making: xfce4-docklike-plugin-git
```

For the record, I also got this error when compiling from git but using `libtoolize && aclocal && autoconf && automake && make` (suggested [here](https://stackoverflow.com/questions/58565768/libtool-version-mismatch-error-2-4-6-expected-vs-2-4-6-42-b88ce-actual-ac)) fixes it.

**EDIT:** [Here](https://pastebin.com/hSpQLEFa) is the complete output of all the commands

--- a/src/Dock.hpp
+++ b/src/Dock.hpp
@@ -7,11 +7,7 @@
 #ifndef TASKBAR_HPP
 #define TASKBAR_HPP
 
-extern "C"
-{
-#include <libxfce4panel/libxfce4panel.h>
-#include <libxfce4ui/libxfce4ui.h>
-}
+
 
 #include <gtk/gtk.h>
 #include <libwnck/libwnck.h>

--- a/src/Dock.hpp
+++ b/src/Dock.hpp
@@ -7,8 +7,6 @@
 #ifndef TASKBAR_HPP
 #define TASKBAR_HPP
 
-
-
 #include <gtk/gtk.h>
 #include <libwnck/libwnck.h>
 
@@ -21,6 +19,7 @@
 #include "Settings.hpp"
 #include "Store.tpp"
 #include "Wnck.hpp"
+
 class Group;
 
 namespace Dock

--- a/src/SettingsDialog.hpp
+++ b/src/SettingsDialog.hpp
@@ -14,13 +14,6 @@
 #include "Helpers.hpp"
 #include "Settings.hpp"
 
-extern "C"
-{
-#include <libxfce4panel/libxfce4panel.h>
-#include <libxfce4ui/libxfce4ui.h>
-#include <libxfce4util/libxfce4util.h>
-}
-
 #include "Plugin.hpp"
 
 namespace SettingsDialog

